dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/ProgramInstance.java

```diff
@@ -188,6 +188,19 @@ public ProgramStageInstance getActiveProgramStageInstance()
 
         return null;
     }
+    
+    public boolean hasActiveProgramStageInstance( ProgramStage programStage )

```

Comment:
How performant is this? what if there are 100k program stage instances? (single event?) maybe better to use a HQL query or something?

Reply:
Ok, should probably be the same then... as long as its with registration and for non repeatable only

Reply:
yea.